protos/feast/core/Registry.proto

```diff
@@ -0,0 +1,35 @@
+//
+// * Copyright 2020 The Feast Authors
+// *
+// * Licensed under the Apache License, Version 2.0 (the "License");
+// * you may not use this file except in compliance with the License.
+// * You may obtain a copy of the License at
+// *
+// *     https://www.apache.org/licenses/LICENSE-2.0
+// *
+// * Unless required by applicable law or agreed to in writing, software
+// * distributed under the License is distributed on an "AS IS" BASIS,
+// * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// * See the License for the specific language governing permissions and
+// * limitations under the License.
+//
+
+syntax = "proto3";
+
+package feast.core;
+option java_package = "feast.proto.core";
+option java_outer_classname = "RegistryProto";
+option go_package = "github.com/feast-dev/feast/sdk/go/protos/feast/core";
+
+import "feast/core/Entity.proto";
+import "feast/core/FeatureTable.proto";
+import "google/protobuf/timestamp.proto";
+
+message Registry {
+    repeated Entity entities = 1;
+    repeated FeatureTable feature_tables = 2;
+
+    string registry_schema_version = 3; // to support migrations
+    string version = 4; // version id, random string generated on each update
```

Comment:
What problem does this random string solve, and can it be replaced with a hash of the objects?

Reply:
It could be a hash, random string is just easier to implement. Properly hashing protos isn't super straightforward

Reply:
How will this version be used?

Reply:
Just debug info for now. For example I would maybe propagate this to tags when creating cloud resources. It is generally nice to have a notion of a "version" for the registry state

Reply:
Ok, so it's almost closer to an id?

Reply:
I see, maybe `version_id` would be a clearer name then?